### PR TITLE
bpo-33649: Cleanup asyncio/streams and asyncio/synchronization docs sections

### PR DESCRIPTION
**Pull request by 1st1**
https://bugs.python.org/issue33649

### CODE REVIEW

File: Doc/library/asyncio-stream.rst [hunk 1]
```diff
@@ -136,67 +143,47 @@ and work with streams:
...
    .. coroutinemethod:: readline()
 
-      Read one line, where "line" is a sequence of bytes ending with ``\n``.
+      Read one line, where "line" is a sequence of bytes
+      ending with ``\n``.
 
-      If EOF is received, and ``\n`` was not found, the method will
-      return the partial read bytes.
+      If an EOF is received and ``\n`` was not found, the method
+      returns a partially read data.
```
Comment (willingc) [inline Doc/library/asyncio-stream.rst]: ~a~

File: Doc/library/asyncio-sync.rst [hunk 2]
```diff
@@ -1,305 +1,324 @@
...
 .. class:: Lock(\*, loop=None)
 
-   Primitive lock objects.
+   Implements a mutex lock for asyncio tasks.  Not thread-safe.
 
-   A primitive lock is a synchronization primitive that is not owned by a
-   particular coroutine when locked.  A primitive lock is in one of two states,
-   'locked' or 'unlocked'.
+   An asyncio lock can be used to guarantee exclusive access to a
+   shared resource to a single asyncio Task.
```
Comment (willingc) [inline Doc/library/asyncio-sync.rst]: ~to~ of

**Review by asvetlov — APPROVED** [review]
lgtm

File: Doc/library/asyncio-sync.rst [hunk 3]
```diff
@@ -1,305 +1,324 @@
...
-   The lock is created in the unlocked state.
-   It has two basic methods, :meth:`acquire` and :meth:`release`.
-   When the state is unlocked, acquire() changes the state to
-   locked and returns immediately.  When the state is locked, acquire() blocks
-   until a call to release() in another coroutine changes it to unlocked, then
-   the acquire() call resets it to locked and returns.  The release() method
-   should only be called in the locked state; it changes the state to unlocked
-   and returns immediately.  If an attempt is made to release an unlocked lock,
-   a :exc:`RuntimeError` will be raised.
+   The preferred way to use a Lock is an :keyword:`async with`
```
Comment (willingc) [inline Doc/library/asyncio-sync.rst:42]: Perhaps have the example code after introducing: acquire, lock, release, locked.

File: Doc/library/asyncio-sync.rst [hunk 4]
```diff
@@ -1,305 +1,323 @@
...
+   :meth:`clear` method.  The :meth:`wait` method blocks until the
+   flag is set to *true*.  The flag is set to *false* initially.
 
-   .. method:: clear()
+   Example::
 
-      Reset the internal flag to false. Subsequently, coroutines calling
-      :meth:`wait` will block until :meth:`set` is called to set the internal
-      flag to true again.
+      async def waiter(e):
```
Comment (willingc) [inline Doc/library/asyncio-sync.rst]: Perhaps add a short docstring to the example: `e` is not defined for the reader until further down the page.

File: Doc/library/asyncio-sync.rst [hunk 5]
```diff
@@ -1,305 +1,323 @@
...
-   finally:
-       lock.release()
 
 .. deprecated:: 3.7
 
-   Lock acquiring using ``await lock`` or ``yield from lock`` and
+   Acquiring a lock using ``await lock`` or ``yield from lock`` and/or
    :keyword:`with` statement (``with await lock``, ``with (yield from
-   lock)``) are deprecated.
+   lock)``) is deprecated.
```
Comment (willingc) [inline Doc/library/asyncio-sync.rst]: Use `async with lock` instead.

**Review by willingc — COMMENTED** [review]
A few minor typos/corrections..

File: Doc/library/asyncio-sync.rst (hunk 2 above)
Comment (1st1) [reply Doc/library/asyncio-sync.rst]: "of a single asyncio Task."?  We use locks to guarantee access *(access to a shared resource)* to an asyncio task.  With "of" I'm not sure it reads correctly...

File: Doc/library/asyncio-sync.rst (hunk 3 above)
Comment (1st1) [reply Doc/library/asyncio-sync.rst:42]: Yeah, I thought about it too.  If we do that the examples won't be immediately visible to the reader.  OTOH acquire and release are so wide known that understanding these short examples (read "usage recommendations") shouldn't be a problem...

File: Doc/library/asyncio-sync.rst (hunk 4 above)
Comment (1st1) [reply Doc/library/asyncio-sync.rst]: I'll just give proper names to variables and arguments; using one-letter names is indeed very confusing.

File: Doc/library/asyncio-sync.rst (hunk 3 above)
Comment (willingc) [reply Doc/library/asyncio-sync.rst:42]: Good point.

File: Doc/library/asyncio-sync.rst (hunk 2 above)
Comment (willingc) [reply Doc/library/asyncio-sync.rst]: Maybe: "to a single asyncio Task and its shared resource.

Comment (1st1) [reply Doc/library/asyncio-sync.rst]: The problem is that the resource (typically) does not belong to the Task -- the Task just wants to exclusively work with the resource for some time; hence "**its** shared resource." isn't 100% correct.

Maybe we should just drop the "to a single asyncio Task" part:

```
An asyncio lock can be used to guarantee exclusive access to a shared resource.
```

?

Comment (willingc) [reply Doc/library/asyncio-sync.rst]: I like that. Dropping the phrase is better.

**Review by willingc — APPROVED** [review]
Thanks @1st1 and @asvetlov.

**Pull request merged**